packages/ui-keyring/src/index.ts

```diff
@@ -4,83 +4,429 @@
+let hasCalledInitOptions = false;
 
 // FIXME The quicker we get in https://github.com/polkadot-js/apps/issues/138
 // the better, this is now completely out of control
-export default ({
-  addAccountPair: (pair: KeyringPair, password: string): KeyringPair =>
-    addAccountPair(state, pair, password),
-  backupAccount: (pair: KeyringPair, password: string): KeyringPair$Json =>
-    backupAccount(state, pair, password),
-  createAccount: (seed: Uint8Array, password?: string, meta?: KeyringPair$Meta): KeyringPair =>
-    createAccount(state, seed, password, meta),
-  createAccountMnemonic: (seed: string, password?: string, meta?: KeyringPair$Meta): KeyringPair =>
-    createAccountMnemonic(state, seed, password, meta),
-  encryptAccount: (pair: KeyringPair, password: string): void =>
-    encryptAccount(state, pair, password),
-  forgetAccount: (address: string): void =>
-    forgetAccount(state, address),
-  forgetAddress: (address: string): void =>
-    forgetAddress(state, address),
-  isAvailable: (address: string | Uint8Array): boolean =>
-    isAvailable(state, address),
-  isPassValid: (password: string): boolean =>
-    isPassValid(state, password),
-  getAccounts: (): Array<KeyringAddress> =>
-    getAccounts(state),
-  getAddress: (address: string | Uint8Array): KeyringAddress =>
-    getAddress(state, address),
-  getAddresses: (): Array<KeyringAddress> =>
-    getAddresses(state),
-  getPair: (address: string | Uint8Array): KeyringPair =>
-    state.keyring.getPair(address),
-  getPairs: (): Array<KeyringPair> =>
-    state.keyring.getPairs().filter((pair) =>
-      development.isDevelopment() || pair.getMeta().isTesting !== true
-    ),
-  loadAll: (): void =>
-    loadAll(state),
-  restoreAccount: (json: KeyringPair$Json, password: string): KeyringPair =>
-    restoreAccount(state, json, password),
-  saveAccount: (pair: KeyringPair, password?: string): void =>
-    saveAccount(state, pair, password),
-  saveAccountMeta: (pair: KeyringPair, meta: KeyringPair$Meta): void =>
-    saveAccountMeta(state, pair, meta),
-  saveAddress: (address: string, meta: KeyringPair$Meta): void =>
-    saveAddress(state, address, meta),
-  saveRecent: (address: string): SingleAddress =>
-    saveRecent(state, address),
-  setDevMode: (isDevelopment: boolean): void =>
-    development.set(isDevelopment)
-} as KeyringInstance);
+class Keyring implements KeyringInstance {
+  private state: State;
+  private id: number = 0;
+
+  static counter: number;
+
+  constructor () {
+    // state must be always defined in the constructor even if trying to create second instance to overcome TSLint error
+    this.state = {
+      accounts: observableAccounts,
+      addresses: observableAddresses,
+      keyring: testKeyring()
+    };
+
+    if (!singletonInstance) {
+      singletonInstance = this;
+    } else {
+      return;
+    }
+
+    // NOTE Everything is loaded in API after chain is received
+    // this.loadAll();
+
+    this.id = ++Keyring.counter;
```

Comment:
All this should not been needed. Since we don't export `class Keyring` and rather export the only instance we create, there will only be one in existence. So basically,

```
// not exported
class Keyring {
...
}

// since instance
const keyring = new Keyring();

// export the only instance ever
export default keyring;
```

would do the trick